modules/caddyhttp/metrics.go

```diff
@@ -138,6 +185,12 @@ func (h *metricsInstrumentedHandler) ServeHTTP(w http.ResponseWriter, r *http.Re
 		statusLabels["host"] = strings.ToLower(r.Host)
 	}
 
+	customLabels := h.processCustomLabels(r)
+	for key, value := range customLabels {
+		labels[key] = value
+		statusLabels[key] = value
+	}
+
```

Comment:
What I'm afraid of, happened. I've just tested and all the replaced labels have the value "unknown". Applying replacements here is too early because many of the values in the replacer aren't filled in yet. Can this be moved lower or refactored so it picks up the labeling? I imagine it needs something similar to `observeRequest`.

Reply:
can you tell me the labels key that is being recorded as unknown at that point? i thought the only value net yet determined at that point was `status`, and that `status` is determined in the code below, so i thought there was no problem.
but i'm not familiar with this code base, so if there's something i haven't discovered yet, plz let me know for reproducing test.